doc/user-guide/data-structures.rst

```diff
@@ -495,6 +497,314 @@ dimension and non-dimension variables:
+:py:class:`~xarray.DataTree` objects.
+
+.. ipython:: python
+
+    dt["child-node"]
+    dt["foo"]
+
+Iterating over keys will iterate over both the names of variables and child nodes.
+
+We can also access all the data in a single node, and its inherited coordinates, through a dataset-like view
+
+.. ipython:: python
+
+    dt["child-node"].dataset
+
+This demonstrates the fact that the data in any one node is equivalent to the
+contents of a single :py:class:`~xarray.Dataset` object. The :py:attr:`DataTree.dataset <xarray.DataTree.dataset>` property
+returns an immutable view, but we can instead extract the node's data contents
+as a new and mutable :py:class:`~xarray.Dataset` object via
+:py:meth:`DataTree.to_dataset() <xarray.DataTree.to_dataset>`:
+
+.. ipython:: python
+
+    dt["child-node"].to_dataset()
+
+This same call can be made to get only the local node variables without any
+inherited ones, by setting the inherited keyword to False, but in this example
+there are no inherited coordinates so the result is the same as the previous call.
+
+.. ipython:: python
+
+    dt["child-node"].to_dataset(inherited=False)
+
+
+Like with :py:class:`~xarray.Dataset`, you can access the data and coordinate variables of a
+node separately via the :py:attr:`~xarray.DataTree.data_vars` and :py:attr:`~xarray.DataTree.coords` attributes:
+
+.. ipython:: python
+
+    dt["child-node"].data_vars
+    dt["child-node"].coords
+
+
+Dictionary-like methods
+~~~~~~~~~~~~~~~~~~~~~~~
+
+We can update a datatree in-place using Python's standard dictionary syntax,
+similar to how we can for Dataset objects. For example, to create this example
+DataTree from scratch, we could have written:
+
+.. ipython:: python
+
+    dt = xr.DataTree(name="root")
+    dt["foo"] = "orange"
+    dt["child-node"] = xr.DataTree(
+        dataset=xr.Dataset({"bar": 0}, coords={"y": ("y", [0, 1, 2])})
+    )
+    dt["child-node/new-zed-node/zed"] = np.nan
+    dt
+
+To change the variables in a node of a :py:class:`~xarray.DataTree`, you can use all the
+standard dictionary methods, including ``values``, ``items``, ``__delitem__``,
+``get`` and :py:meth:`xarray.DataTree.update`.
+Note that assigning a :py:class:`~xarray.DataTree` object to a :py:class:`~xarray.DataTree` variable using
+``__setitem__`` or :py:meth:`~xarray.DataTree.update` will :ref:`automatically align <update>` the
+array(s) to the original node's indexes.
+
+If you copy a :py:class:`~xarray.DataTree` using the :py:func:`copy` function or the
+:py:meth:`xarray.DataTree.copy` method it will copy the subtree,
+meaning that node and children below it, but no parents above it.
+Like for :py:class:`~xarray.Dataset`, this copy is shallow by default, but you can copy all the
+underlying data arrays by calling ``dt.copy(deep=True)``.
+
+
+.. _datatree-inheritance:
+
+DataTree Inheritance
+~~~~~~~~~~~~~~~~~~~~
+
+DataTree implements a simple inheritance mechanism. Coordinates and their
```

Comment:
For clarity (this has come up a few times recently between me and Tom), I think it's good to emphasize that dimensions are inherited too (even if there is not an associated coordinate):
```suggestion
DataTree implements a simple inheritance mechanism. Coordinates, dimensions and their
```

Reply:
See https://github.com/pydata/xarray/issues/9463 and https://github.com/pydata/xarray/pull/9470#issuecomment-2342335745.